src/pkg/services/m365/api/lists.go

```diff
@@ -258,8 +264,8 @@ func (c Lists) PostList(
 
 	if newList != nil &&
 		newList.GetList() != nil &&
-		ptr.Val(newList.GetList().GetTemplate()) == WebTemplateExtensionsListTemplateName {
-		return nil, clues.StackWC(ctx, ErrCannotCreateWebTemplateExtension)
+		NonRestorableListTemplates.HasKey(ptr.Val(newList.GetList().GetTemplate())) {
```

Comment:
nit: `newList != nil` is not really needed since `ToListable` always creates a valid `newList`.